osgi/src/main/java/com/hazelcast/osgi/examples/SampleActivator.java

```diff
@@ -0,0 +1,69 @@
+package com.hazelcast.osgi.examples;
+
+import com.hazelcast.config.Config;
+import com.hazelcast.osgi.HazelcastOSGiInstance;
+import com.hazelcast.osgi.HazelcastOSGiService;
+import org.osgi.framework.BundleActivator;
+import org.osgi.framework.BundleContext;
+import org.osgi.framework.ServiceReference;
+
+/**
+ * {@link org.osgi.framework.BundleActivator} implementation for OSGI sample which is activated by our sample bundle.
+ */
+public class SampleActivator implements BundleActivator {
+
+    @Override
+    public void start(BundleContext context) throws Exception {
+        final ClassLoader currentContextClassLoader = Thread.currentThread().getContextClassLoader();
+        try {
+            Thread.currentThread().setContextClassLoader(getClass().getClassLoader());
+
+            System.out.println("Starting activator " + this + " in bundle " + context.getBundle() + " ...");
+
+            // Find the service reference of `HazelcastOSGiService` instance
+            ServiceReference serviceRef =
+                    context.getServiceReference(HazelcastOSGiService.class.getName());
+            if (serviceRef == null) {
+                throw new IllegalStateException("There is no registered `HazelcastOSGiService`!");
+            }
```

Comment:
If possible using @Service annotation based injection instead would also be nice (maybe as a separate example?), as it's quite widespread.